File: CS/SunburstHierarchicalDataAdapterSample/DataObjects.cs
using System.Collections.Generic;

namespace SunburstHierarchicalDataAdapterSample {
    public class CountryInfo {
        public string CountryName { get; set; }
        public List<CityInfo> CityInfos { get; set; } = new List<CityInfo>();
    }
    public class CityInfo {
        public string CityName { get; set; }
        public List<ProductInfo> SaleInfos { get; set; } = new List<ProductInfo>();
    }
    public class ProductInfo {
        public string Category { get; set; }
        public double Total { get; set; }
    }
}

File: VB/SunburstHierarchicalDataAdapterSample/DataObjects.vb
Imports System.Collections.Generic

Namespace SunburstHierarchicalDataAdapterSample
    Public Class CountryInfo
        Public Property CountryName() As String
        Public Property CityInfos() As New List(Of CityInfo)()
    End Class
    Public Class CityInfo
        Public Property CityName() As String
        Public Property SaleInfos() As New List(Of ProductInfo)()
    End Class
    Public Class ProductInfo
        Public Property Category() As String
        Public Property Total() As Double
    End Class
End Namespace
